app/classifier/tasks/survey/editor.cjsx

```diff
@@ -443,10 +443,14 @@ module.exports = createReactClass
       @state.task.characteristics[characteristicID].values[valueID].image = valueIcon
       for choiceID, choice of @state.task.choices
         choice.characteristics[characteristicID] ?= []
-      if isSet
-        choiceID = @makeID row.name
-        if @state.task.choices[choiceID]?.characteristics[characteristicID].indexOf(valueID) is -1
-          @state.task.choices[choiceID]?.characteristics[characteristicID].push valueID
+      choiceID = @makeID row.name
+      choiceCharacteristicArray = @state.task.choices[choiceID]?.characteristics[characteristicID]
+      valueIdIndex = choiceCharacteristicArray.indexOf valueID
+      if isSet and valueIdIndex is -1
+        @state.task.choices[choiceID]?.characteristics[characteristicID]?.push valueID
+      if not isSet and valueIdIndex > -1
+        choiceCharacteristicArray.splice valueIdIndex, 1
+        @state.task.choices[choiceID]?.characteristics[characteristicID] = choiceCharacteristicArray      
```

Comment:
Lines 452 and 453 can likely be combined, don't think `choiceCharacteristicArray` really necessary anywhere. I'll clean up this afternoon, but I think underlying logic ok.